operator/src/main/java/oracle/kubernetes/operator/helpers/ServiceHelper.java

```diff
@@ -528,9 +535,33 @@ private Step createNewService(Step next) {
 
     private Step deleteAndReplaceService(Step next) {
       V1DeleteOptions deleteOptions = new V1DeleteOptions();
+      if (getSpecType().equals(NODE_PORT_TYPE)) {
+        return deleteAndReplaceNodePortService();
+      } else {
+        return new CallBuilder()
+            .deleteServiceAsync(
+                createServiceName(), getNamespace(), getDomainUid(), deleteOptions, new DeleteServiceResponse(next));
+      }
+    }
+
+    private Step deleteAndReplaceNodePortService() {
       return new CallBuilder()
-          .deleteServiceAsync(
-              createServiceName(), getNamespace(), getDomainUid(), deleteOptions, new DeleteServiceResponse(next));
+              .withLabelSelectors(forDomainUidSelector(info.getDomainUid()), getCreatedbyOperatorSelector())
+              .listServiceAsync(
+                      getNamespace(),
+                      new ActionResponseStep<V1ServiceList>() {
+                      public Step createSuccessStep(V1ServiceList result, Step next) {
+                        Collection<V1Service> c = Optional.ofNullable(result).map(list -> list.getItems().stream()
+                                  .filter(s -> isNodePortService(s))
+                                  .collect(Collectors.toList())).orElse(new ArrayList<>());
+                        return new DeleteServiceListStep(c, createReplacementService(next));
+                      }
+
+                      private boolean isNodePortService(V1Service svc) {
```

Comment:
Can we use the existing isNodePortType method instead of introducing a new method?

Reply:
Thanks. For some reason, I thought it was a no-arg method. I have made the suggested change.